### PR TITLE
refactoring to use php8-ga

### PR DESCRIPTION
**Pull request by fabiocicerchia**

### CODE REVIEW

**Review by JellyBellyDev — APPROVED** [review]
LGTM

File: src/Contributors/Model/Contributor.php [hunk 1]
```diff
@@ -39,7 +39,8 @@ public function getProfileUrl(): string
 
     public function getProfileImg(): string
     {
-        return $this->profileImg.'?s='.$this->size;
+        $sep = strpos($this->profileImg, '?') ? '&' : '?';
```
Comment (AlessandroMinoccheri) [inline src/Contributors/Model/Contributor.php:42]: Is it possible to simplify this line to have a more readable code? Maybe split into more lines or create a separate function. What do you think?

**Comment by JellyBellyDev** [comment]
to pass the ci you need to modify the doker image used by circle

**Comment by JellyBellyDev** [comment]
@fabiocicerchia i add use the new docker image to circleci.

**Comment by JellyBellyDev** [comment]
@fabiocicerchia there are still many dependencies not compatible with php8, sure you have updated them all?

File: src/Badge/Model/Package.php [hunk 2]
```diff
@@ -62,7 +62,7 @@ public function getPackageDownloads(?string $type = 'total'): ?int
 
     private function comparator(Version $version1, Version $version2): bool
     {
-        return $version1->getTime() > $version2->getTime();
+        return $version1->getTime() <=> $version2->getTime();
```
Comment (JellyBellyDev) [inline src/Badge/Model/Package.php:65]: this operator return a float and not a bool. wrong?

File: src/Contributors/Model/Contributor.php [hunk 3]
```diff
@@ -37,9 +37,18 @@ public function getProfileUrl(): string
...
+               ? '&'
+               : '?';
+    }
+
     public function getProfileImg(): string
     {
-        return $this->profileImg;
+        $sep = $this->getUrlSeparator($this->profileImg);
+
+        return $this->profileImg.$sep.'s='.$this->size;
```
Comment (JellyBellyDev) [inline src/Contributors/Model/Contributor.php:51]: WDYT to use `http_build_query `?

File: sys/nginx/default.conf.template [hunk 4]
```diff
@@ -75,7 +75,8 @@ server {
             return string.gsub(ngx.var.document_uri, "/dark%-canary", "")
         }
 
-        fastcgi_pass ${PHPFPM_CANARY_HOST}:9001;
+        # fastcgi_pass ${PHPFPM_CANARY_HOST}:9001;
+        fastcgi_pass ${PHPFPM_CANARY_HOST}:9000;
```
Comment (JellyBellyDev) [inline sys/nginx/default.conf.template:79]: which of the two?

**Review by garak — APPROVED** [review]

**Review by AlessandroMinoccheri — APPROVED** [review]
LGTM 👍

**Pull request merged**